org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/ReferenceBinding.java

```diff
@@ -525,6 +526,14 @@ else if (CharOperation.equals(TypeConstants.JDK_INTERNAL_PREVIEW_FEATURE, this.c
 							return;
 					}
 					return;
+				case 7: // jakarta
+					switch (packageName[1]) {
+						case 'a':
+							if (CharOperation.equals(TypeConstants.JAKARTA_ANNOTATION_INJECT_INJECT, this.compoundName))
+								this.id = TypeIds.T_JavaxInjectInject;
```

Comment:
Can you confirm that `javax.annotation.inject.Inject` and `jakarta.annotation.inject.Inject` are indeed *identical* from a semantic p.o.v.? Otherwise, a new constant in `TypeIds` might be better.

Reply:
`jakarta.inject.Inject` is a 1:1 replacement of `javax.inject.Inject`. Therefore the semantics are identical.